docs/using-keystone-webhook-authenticator-and-authorizer.md

```diff
@@ -334,53 +410,111 @@ and both as well, depending on your requirement.
 
 ### Configuration on K8S master for authentication and authorization
 
-- Create webhook config file. `10.109.16.219` is the cluster IP of
-  k8s-keystone-auth service. We reuse the folder `/etc/kubernetes/pki/` because
-  it's already mounted and accessible by API server pod.
-
-  ```shell
-  cat <<EOF > /etc/kubernetes/pki/webhookconfig.yaml
-  ---
-  apiVersion: v1
-  kind: Config
-  preferences: {}
-  clusters:
-    - cluster:
-        insecure-skip-tls-verify: true
-        server: https://10.109.16.219:8443/webhook
-      name: webhook
-  users:
-    - name: webhook
-  contexts:
-    - context:
-        cluster: webhook
-        user: webhook
-      name: webhook
-  current-context: webhook
-  EOF
-  ```
-
-- Modify API server config file to use the webhook service for authentication.
-
-  ```shell
-  sed -i '/image:/ i \ \ \ \ - --authentication-token-webhook-config-file=/etc/kubernetes/pki/webhookconfig.yaml' /etc/kubernetes/manifests/kube-apiserver.yaml
-  ```
-
-- Modify API server config file to use the webhook service for authorization.
-
-  ```shell
-  sed -i '/image:/ i \ \ \ \ - --authorization-webhook-config-file=/etc/kubernetes/pki/webhookconfig.yaml' /etc/kubernetes/manifests/kube-apiserver.yaml
-  sed -i "/authorization-mode/c \ \ \ \ - --authorization-mode=Node,Webhook,RBAC" /etc/kubernetes/manifests/kube-apiserver.yaml
-  ```
-
-- Wait for the API server to restart successfully until you can get all the
-  pods in `kube-system` namespace by running `kubectl get pod -n kube-system`
+- Create webhook config file. We reuse the folder `/etc/kubernetes/pki/`
+  because it's already mounted and accessible by API server pod.
+
+    ```shell
+    cat <<EOF > /etc/kubernetes/pki/webhookconfig.yaml
+    ---
+    apiVersion: v1
+    kind: Config
+    preferences: {}
+    clusters:
+      - cluster:
+          insecure-skip-tls-verify: true
+          server: https://${keystone_auth_service_addr}:8443/webhook
+        name: webhook
+    users:
+      - name: webhook
+    contexts:
+      - context:
+          cluster: webhook
+          user: webhook
+        name: webhook
+    current-context: webhook
+    ```
+
+- Modify kube-apiserver config file to use the webhook service for
+  authentication and authorization.
+
+    ```shell
+    sed -i '/image:/ i \ \ \ \ - --authentication-token-webhook-config-file=/etc/kubernetes/pki/webhookconfig.yaml' /etc/kubernetes/manifests/kube-apiserver.yaml
+    sed -i '/image:/ i \ \ \ \ - --authorization-webhook-config-file=/etc/kubernetes/pki/webhookconfig.yaml' /etc/kubernetes/manifests/kube-apiserver.yaml
+    sed -i "/authorization-mode/c \ \ \ \ - --authorization-mode=Node,Webhook,RBAC" /etc/kubernetes/manifests/kube-apiserver.yaml
+    ```
+
+- Wait for the API server to restart successfully until you can see all the
+  pods are running in `kube-system` namespace.
+
+## Authorization policy definition(version 2)
```

Comment:
We should have discussions on version2. Too many designs and details should be talked.